### PR TITLE
feat: use dash notation for inputs (deprecates underscore notation)

### LINKED ISSUE #57: Proposal: Use dashes (instead of underscores) in input names
From @gr2m in https://github.com/actions/create-github-app-token/pull/54#discussion_r1344571256:
> I think we should use dashes for our arguments, it seems to be the convention for `actions/*` actions, e.g. see https://github.com/actions/checkout/

Notably, changing the input names affects the translated environment variable names[^1]:
- An input named `my_var` is translated to `INPUTS_MY_VAR`.
- An input named `my-var` is translated to `INPUTS_MY-VAR`.

So, we may want a plan for backwards-compatibility, or we may want to ship this in a major version bump.

[^1]: cf. [“Metadata syntax for GitHub Actions > Example: Specifying inputs”](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#example-specifying-inputs)

### PR DESCRIPTION
**Pull request by smockle**
Fixes #57 

This PR implements the 3-step plan proposed by @gr2m in https://github.com/actions/create-github-app-token/issues/57#issuecomment-1751272252:

> 1. Support both input types
> 2. Log a deprecation warning for the old notation
> 3. Add a test for deprecations

Although this PR supports both input formats simultaneously, I opted _not_ to document the old format in the updated README. That’s a decision I’m happy to revisit, if y’all would prefer to have documentation for both the old and new formats.

### CODE REVIEW

**Comment by parkerbxyz** [comment]
This is great. Thanks, @smockle! ✨ 

> Although this PR supports both input formats simultaneously, I opted _not_ to document the old format in the updated README.

I agree with this approach. 🙂

File: action.yml [hunk 1]
```diff
@@ -5,21 +5,33 @@ branding:
...
   color: "gray-dark"
 inputs:
+  app-id:
+    description: "GitHub App ID"
+    required: false # TODO: When 'app_id' is removed, make 'app-id' required
   app_id:
     description: "GitHub App ID"
-    required: true
+    required: false
+    deprecationMessage: "'app_id' is deprecated and will be removed in a future version. Use 'app-id' instead."
```
Comment (gr2m) [inline action.yml:14]: ohhh nice, today I learned, very cool!
https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputsinput_iddeprecationmessage

**Review by gr2m — APPROVED** [review]
wow that was quick, looks good to me 👍🏼

File: action.yml (hunk 1 above)
Comment (parkerbxyz) [reply action.yml:14]: Same!

**Comment by gr2m** [comment]
interesting side effect of this PR: the release build failed because it's using the stable version:
https://github.com/actions/create-github-app-token/actions/runs/6436042980

We should use the local version instead, like in tests

**Pull request merged**